### ISSUE TITLE
Control.Concurrent.Chan deprecated

### ISSUE THREAD

**Issue by RyanGlScott** [ku-fpg/hermit-web#3]
When compiling HERMIT-Web, a warning appears about <tt>isEmptyChan</tt> being deprecated:

```
[4 of 5] Compiling HERMIT.Web.Actions ( src/HERMIT/Web/Actions.hs, dist/build/HERMIT/Web/Actions.o )

src/HERMIT/Web/Actions.hs:106:27: Warning:
    In the use of `isEmptyChan'
    (imported from Control.Concurrent.Chan):
    Deprecated: "if you need this operation, use Control.Concurrent.STM.TChan instead.  See http://hackage.haskell.org/trac/ghc/ticket/4154 for details"
```

Is it safe to replace Chan with STM.TChan?

**Comment by xich** [comment]
Yes, it's safe, but not a straightforward replacement, as you have to plumb some additional transaction runners. I'll do this eventually.

**Comment by RyanGlScott** [comment]
Closed in fa1602e0054d157fe1c61707e40aef2874f58c9b.
